### ISSUE TITLE
Add content-build and Drupal cache into existing FE Docker image

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#51937]
# Description
We are currently exploring 5 scenarios for a FE image workflow:
https://vfs.atlassian.net/wiki/spaces/TT4/pages/2546630657/FE+Image+Scenarios

# Tasks
- [x] Detail tickets for each of the options

# Acceptance Criteria
- [ ] The FE Docker image runs both `vets-website` and `content-build` through one or more of these approaches

**Comment by cyriltech** [comment]
Modified existing Dockerfile to [this](https://github.com/department-of-veterans-affairs/vets-website/blob/new-docker/Dockerfile1) as a starting point for re-building.

**Comment by cyriltech** [comment]
I follow the [steps](https://depo-platform-documentation.scrollhelp.site/developer-docs/setting-up-your-local-frontend-environment) for running front-end locally, I was able to see vets-website running, fetching drupal cache took about an hr to load up content then exited with the error from attached file![front-end doc.jpg](https://images.zenhubusercontent.com/6385292bddbc0cbb14b84a51/3ddb0579-65e6-45ca-81ae-341ce35ab66f)1.

**Comment by cyriltech** [comment]
Also, modified existing Dockerfile to [this](https://github.com/department-of-veterans-affairs/vets-website/blob/new-docker/Dockerfile1) then performed a docker build that took about 4hrs to fetch drupal cache and completed with attached error at the end![docker-build.jpg](https://images.zenhubusercontent.com/6385292bddbc0cbb14b84a51/065449ab-2058-4809-be9f-407bb7101a51). Not connected Socks at the moment, next move is to get socks on my new device and see what happens fetching drupal cache with socks.

**Comment by cyriltech** [comment]
After trying to add content build severally, I decided to reach out for help from the team managing Drupal. [Here](https://dsva.slack.com/archives/C04CYC4LMU6/p1675714448471609) is a thread of ongoing collaboration.

**Comment by pjhill** [comment]
I believe [this script](https://github.com/department-of-veterans-affairs/content-build/blob/main/script/drupal-aws-cache.js) exists explicitly to facilitate with retrieving the Drupal content.

There should be `yarn` commands that call this script with some helpers. Have you tried something like `yarn fetch-drupal-cache`?
![image.png](https://images.zenhubusercontent.com/5d2633e0a6db635dbcbaf8d5/da274bed-12ee-44ec-94db-dee1c9862f8a)

**Comment by pjhill** [comment]
Revisiting the documentation, I think you're going to be "inside" the VA network to be able to access the Drupal cache. That's going to make local testing of the Dockerfile difficult, and I'm not entirely sure that the GHA runner that picks up the job to build, tag, and store this Docker image in ECR will be able to access the Drupal cache.

I believe if we use self hosted GHA runners that MAY be the case. The next piece of research we probably need to do is to determine whether self-hosted runners will be able to retrieve the Drupal cache.

I'm thinking something like this has to be the case in order for the production deployment workflow to succeed.

**Comment by pjhill** [comment]
Oddly, I think I was just able to retrieve the Drupal cache without being connected to SOCKS or otherwise having routes to "internal" resources:
![image.png](https://images.zenhubusercontent.com/5d2633e0a6db635dbcbaf8d5/3d0d1e1a-42df-4dc1-9895-f472773f2319)

**Comment by pjhill** [comment]
Let's consider lopping this ticket off at a `content-build` deploy that skips fetching the Drupal cache. We can work on a solution to the Drupal cache in the tickets created here --
* [Research](https://app.zenhub.com/workspaces/platform-tech-team-4-633b069d2920b776613c93d8/issues/gh/department-of-veterans-affairs/va.gov-team/53452)
* [Plan](https://app.zenhub.com/workspaces/platform-tech-team-4-633b069d2920b776613c93d8/issues/gh/department-of-veterans-affairs/va.gov-team/53453)
* [Implement](https://app.zenhub.com/workspaces/platform-tech-team-4-633b069d2920b776613c93d8/issues/gh/department-of-veterans-affairs/va.gov-team/53455)

**Comment by cyriltech** [comment]
Facing some issues using self-hosted runner, at the moment feels like the right configurations should solve the issue that is related to SSL certificate for self-hosted runner. I tried a couple of things with the help of Peter and Curt. Will look more within the Devops Cops to see how to setup a self-hosted runner correctly. here is a link to my workflow  https://github.com/department-of-veterans-affairs/vets-website/actions/runs/4177896608/jobs/7236030415

**Comment by cyriltech** [comment]
![image.png](https://images.zenhubusercontent.com/6385292bddbc0cbb14b84a51/5247ad85-be0c-46ba-98d1-044a35c5591b) I tried to switch the runner to ubuntu-latest, with that I can now fetch Drupal but running into Disc space issue

**Comment by cyriltech** [comment]
I did some reading of the [readme](https://github.com/department-of-veterans-affairs/content-build#readme) for the content-build repo. After that I decided to comment the step "yarn build' on my Dockerfile and use just "yarn fetch-drupal-cache". With this I am able to [ build and push](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/4201038453/jobs/7287646404). I tested the Dockerfile locally, I was able to build and image and run it locally. Here is a a page that is serving at the port 3002 from content-build![content-build.jpg](https://images.zenhubusercontent.com/6385292bddbc0cbb14b84a51/08742a60-3ae4-4087-97e6-10d83ef3f511) 

However, I was not able to see vets-website at 3001 and after executing docker run, it did try to apply similar 56 steps and got an error at 54.
